ladybug/datatype/base.py

```diff
@@ -345,3 +353,69 @@ def ToString(self):
     def __repr__(self):
         """Return Ladybug data type as a string."""
         return self.name
+
+
+class _DataTypeEnumeration(object):
+    """Enumerates all data types, base types, and units."""
+    _TYPES = {}
+    _BASETYPES = {}
+    _UNITS = {}
+    _GENERICTYPE = None
+
+    def __init__(self, import_modules=True):
+        if import_modules is True:
+            self._import_modules()
+
+        for clss in DataTypeBase.__subclasses__():
+            if clss.__name__ != 'GenericType':
+                self._TYPES[clss.__name__] = clss
+                self._BASETYPES[clss.__name__] = clss
+                self._UNITS[clss.__name__] = clss._units
+                for subclss in self._all_subclasses(clss):
+                    self._TYPES[subclss.__name__] = subclss
+            else:
+                self._GENERICTYPE = clss
+
+    @property
+    def types(self):
+        """A tuple indicating all curently supported data types."""
```

Comment:
Currently seems to be your version of "deterioration"! :) missing an r here too.